### ISSUE TITLE
RFE: setsebool: add option to force even if SELinux detected as disabled

### ISSUE THREAD

**Issue by dustymabe** [SELinuxProject/selinux#123]
Hi 

Over in https://github.com/projectatomic/rpm-ostree/issues/1634#issuecomment-435961951 we have identified a need to be able to force setsebool to continue even if selinux appears to be disabled.

Quoting Colin from the comment:

*The quick hack I can think of is an environment variable like e.g. SEMANAGE_FORCE_BUILD=offline or so that was interpreted by setsebool and convinced the tool to rebuild policy and set booleans even if it appeared that SELinux was disabled.*

Would it be acceptable to implement such a configuration ENV variable for setsebool?

**Comment by stephensmalley** [comment]
Not sure offhand why setsebool aborts immediately if SELinux is disabled as it could at least apply persistent changes (assuming -P).  On the other hand, I was wondering semanage boolean provides a way to do this already that doesn't depend on SELinux enabled?

**Comment by bachradsusi** [comment]
In this case, semanage would bring unnecessary dependencies into the system and setsebool already uses libsemanage for setting booleans. It looks like the check whether SELinux is enabled is not necessary; or should not be used when -N is used.

**Comment by dustymabe** [comment]
> Not sure offhand why setsebool aborts immediately if SELinux is disabled as it could at least apply persistent changes (assuming -P). On the other hand, I was wondering semanage boolean provides a way to do this already that doesn't depend on SELinux enabled?

I did not know about `semanage boolean`. Thanks! In our particular use case the `setsebool` operations are executing in rpm post scripts so we can't easily switch to `semanage boolean` without changing a bunch of rpms that we don't control. If possible it would be great if we could get some feature/fix into `setsebool` for this.

**Comment by stephensmalley** [comment]
Tested semanage boolean on a SELinux-disabled system and that worked without problems.
Removed the is_selinux_enabled() test from setsebool and it still failed later on on a SELinux-disabled system, so that requires further investigation into how it differs from semanage boolean and why.

**Comment by stephensmalley** [comment]
setsebool doesn't provide a way to specify the policy store, so it presumes that you want to act on the "active" policy, as configured by /etc/selinux/config.  So even if it were to "work" on a SELinux-disabled system, it would presume that whatever /etc/selinux/config has as the SELINUXTYPE is the policy it should modify.  Is that a safe assumption here?

**Comment by dustymabe** [comment]
do the previous comments in this ticket indicate that we do think a feature request for **`setsebool` to support running even when selinux is disabled** is reasonable?

**Comment by rhatdan** [comment]
Would semanage boolean be better?

**Comment by stephensmalley** [comment]
I suggested that earlier but they said it would bring in additional dependencies.  I have posted a patch for setsebool that fixes it to also work if SELinux is disabled (as long as -P is used; if not, it will still fail since it cannot set active booleans when SELinux is disabled of course).  If you read the linked issue from https://github.com/projectatomic/rpm-ostree/issues/1634#issuecomment-435961951, it appears that the problem is that the container-selinux package runs setsebool -P from %post, so maybe that ought to be fixed regardless.
